joomla/joomla-cms
# Make the Tags RouteHelper language aware

### Summary of Changes
While building a multi-lingual site with tags we found out that the mod_tags_popular module was not producing SEF URLs due to missing the language string. When the menu items had their language set to All (*) it worked fine but when changing them to English (en-GB) and Dutch (nl-NL) the links are no longer working. After some investigation I found out that the Tags RouteHelper does not take the current language into account. This fix resolves that.

### Testing Instructions
1. Create a Joomla site with 2 languages (here I assume EN and NL)
2. Create some Tags associated with the English language
3. Create some Tags associated with the Dutch language
4. Create some Articles associated with the English language and select some English Tags
5. Create some Articles associated with the Dutch language and select some Dutch Tags
6. In the English Home menu create a menu entry of the type `List All Tags` and set the language to English
7. In the English Home menu create a menu entry of the type `Compact List of Tagged Items` and set the language to English, select a tag used by the English articles and set `Content Type` to `Articles`
8. In the Dutch Home menu create a menu entry of the type `List All Tags` and set the language to Dutch
9. In the Dutch Home menu create a menu entry of the type `Compact List of Tagged Items` and set the language to Dutch, select a tag used by the Dutch articles and set `Content Type` to `Articles`
10. Create a front-end module of the type `Tags Popular` and set the language to English and assign it to a menu item that is set in step 7
11. Create a front-end module of the type `Tags Popular` and set the language to Dutch and assign it to a menu item that is set in step 9
12. Go the URL that links to the menu item created in step 7
13. Click on an article shown there
14. You should see the module from step 10 on this page
15. Notice that the links in the module are not as expected but they contain `component/tags/tag`
16. Go the URL that links to the menu item created in step 9
17. Click on an article shown there
18. You should see the module from step 11 on this page
19. Notice that the links in the module are not as expected but they contain `component/tags/tag`
20. Apply the patch
21. Redo step 12 - 14
22. You should see the correct URL in the module
23. Redo step 16 - 18
24. You should see the correct URL in the module

### Actual result BEFORE applying this Pull Request
The URLs in the module do not show the correct SEF URL but instead contain `component/tags/tag`


### Expected result AFTER applying this Pull Request
The URLs in the module do show the correct SEF URL according to their menu item


### Documentation Changes Required
None
